File: pimc/benchmark.py
#******************************************************
# SCRIPT is written for benchmarking pi calculation
# in python. Should run with python>3.0
#******************************************************
import random
import time
import math

nThrows,nSuccess = 0,0
then = time.time()
events=int(1e8)
for i in range(events):
    x,y = random.random(),random.random();      # Throw a dart
    nThrows +=1
    if ( (x*x + y*y) <= 1 ):  nSuccess+=1
itime = time.time() - then
print ("PYTHON : pi value = ", 4*nSuccess/float(nThrows) , "  time (sec) = ",itime)

File: pimc/Benchmark.java
//package bench;

import  java.util.Random;

public class Benchmark {

  public void doBenchmark(boolean print){
    int nThrows = 0; int nSuccess = 0;
    double x, y;
    long then = System.nanoTime();
    Random rnd = new Random();
    int events=(int)1e8;
    for (int i = 0; i < events; i++)   {
      //x = Math.random(); y = Math.random();     // Throw a dart
      x = rnd.nextFloat();
      y = rnd.nextFloat();
      nThrows++;
      if ( x*x + y*y <= 1 )  nSuccess++;
    }
    double itime = ((System.nanoTime() - then)/ ((double)1e9));
    if(print==true)
      System.out.printf("JAVA   : pi value =  %.6f time (sec) = %.4f\n", 4*nSuccess/((double) nThrows),itime);
  }

  public static void main(String[] args){
     Benchmark bench = new Benchmark();

     // this is warmup round printout is set to false
     bench.doBenchmark(false);
     bench.doBenchmark(false);
     //--- now benchmark round
     bench.doBenchmark(true);

  }
}
